beacon_node/store/src/hot_cold_store.rs

```diff
@@ -33,14 +38,17 @@ pub struct HotColdDB<E: EthSpec> {
     /// States with slots less than `split.slot` are in the cold DB, while states with slots
     /// greater than or equal are in the hot DB.
     split: RwLock<Split>,
-    /// Number of slots per restore point state in the freezer database.
-    slots_per_restore_point: u64,
+    config: StoreConfig,
     /// Cold database containing compact historical data.
     pub(crate) cold_db: LevelDB<E>,
     /// Hot database containing duplicated but quick-to-access recent data.
     ///
     /// The hot database also contains all blocks.
     pub(crate) hot_db: LevelDB<E>,
+    /// LRU cache of deserialized blocks. Updated whenever a block is loaded.
+    block_cache: RwLock<LruCache<Hash256, BeaconBlock<E>>>,
```

Comment:
I think these two caches could be `Mutex`?

Reply:
So that if there are two concurrent calls for the same object the second one is guaranteed to hit the cache? E.g. if I load state `X`, I acquire the mutex, miss the cache, load from disk, insert into cache. Then the second call for state `X` hits on the first cache check?

I think the `RwLock` might be preferable, because although it's deficient in the above example, it's better in the more common case of loading many distinct objects because it allows parallelism

Reply:
My thinking was that I just couldn't see anywhere `.read()` is called on it (noting that `LruCache.get` mutates) and my understanding is that a `RwLock` where you only use `.write()` is equivalent to a `Mutex`.

Reply:
Oh yes, you're 100% correct. I'll change it to a `parking_lot::Mutex`